src/server/frontend_xwayland/xwayland_wm.cpp

```diff
@@ -134,21 +136,28 @@ void mf::XWaylandWM::start()
     wm_get_resources();
     setup_visual_and_colormap();
 
-    values[0] =
-        XCB_EVENT_MASK_SUBSTRUCTURE_NOTIFY | XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT | XCB_EVENT_MASK_PROPERTY_CHANGE;
-    xcb_change_window_attributes(xcb_connection, xcb_screen->root, XCB_CW_EVENT_MASK, values);
+    std::vector<uint32_t> const attrib_values{
+        XCB_EVENT_MASK_SUBSTRUCTURE_NOTIFY | XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT | XCB_EVENT_MASK_PROPERTY_CHANGE};
+
+    xcb_change_window_attributes(xcb_connection, xcb_screen->root, XCB_CW_EVENT_MASK, attrib_values.data());
 
     xcb_composite_redirect_subwindows(xcb_connection, xcb_screen->root, XCB_COMPOSITE_REDIRECT_MANUAL);
 
-    supported[0] = xcb_atom.net_wm_moveresize;
-    supported[1] = xcb_atom.net_wm_state;
-    supported[2] = xcb_atom.net_wm_state_fullscreen;
-    supported[3] = xcb_atom.net_wm_state_maximized_vert;
-    supported[4] = xcb_atom.net_wm_state_maximized_horz;
-    supported[5] = xcb_atom.net_active_window;
-    xcb_change_property(xcb_connection, XCB_PROP_MODE_REPLACE, xcb_screen->root, xcb_atom.net_supported, XCB_ATOM_ATOM,
-                        32, /* format */
-                        ARRAY_LENGTH(supported), supported);
+    std::vector<xcb_atom_t> const supported{
```

Comment:
When we can switch on C++17 this should be `std::array const supported{…};`.

Reply:
Surely std::array is C++11? We use it lots.

Reply:
`std::array` is C++11. The template deduction guide which lets you go `std::array const supported{…}` is C++17.

We *could* do `std::array<xcb_atom_t, 6> const supported{…}` right now.